src/screencast/pipewire_screencast.c

```diff
@@ -163,11 +162,20 @@ void xdpw_pwr_stream_init(struct xdpw_screencast_instance *cast) {
 		pw_loop_add_event(state->pw_loop, pwr_on_event, cast);
 	logprint(DEBUG, "pipewire: registered event %p", cast->event);
 
-	params[0] = spa_pod_builder_add_object(&b,
+	enum spa_video_format format = xdpw_format_pw_from_wl_shm(cast);
+	enum spa_video_format format_without_alpha =
+		xdpw_format_pw_strip_alpha(format);
+	uint32_t n_formats = 1;
+	if (format_without_alpha != SPA_VIDEO_FORMAT_UNKNOWN) {
+		n_formats++;
+	}
+
+	const struct spa_pod *param = spa_pod_builder_add_object(&b,
 		SPA_TYPE_OBJECT_Format, SPA_PARAM_EnumFormat,
 		SPA_FORMAT_mediaType,       SPA_POD_Id(SPA_MEDIA_TYPE_video),
 		SPA_FORMAT_mediaSubtype,    SPA_POD_Id(SPA_MEDIA_SUBTYPE_raw),
-		SPA_FORMAT_VIDEO_format,    SPA_POD_Id(xdpw_format_pw_from_wl_shm(cast)),
+		SPA_FORMAT_VIDEO_format,    SPA_POD_CHOICE_ENUM_Id(n_formats,
+			format, format_without_alpha),
```

Comment:
Okay, this is why it isn't working any more for BGRA clients. This needs to be:
```c
SPA_FORMAT_VIDEO_format,    SPA_POD_CHOICE_ENUM_Id(
		++n_formats, // where this is incremented because we now have three arguments
		format, // where this indicates a default choice from a list to follow
		format, // where this begins the list of choices
		format_without_alpha), // where this is the last element of the list
```

Reply:
Basically, we were saying that the unstripped format was the default, but then not including it in the list of allowed values, which only contained the stripped version. That allowed it to work in Chromium.

I'm a little surprised that the builder doesn't complain about this.

Reply:
Ah, good catch, thanks!